examples/gcoap/Makefile

```diff
@@ -26,6 +26,10 @@ BOARD_INSUFFICIENT_MEMORY := arduino-duemilanove arduino-leonardo \
 #GCOAP_TOKENLEN = 2
 #CFLAGS += -DGCOAP_TOKENLEN=$(GCOAP_TOKENLEN)
 
+# Increase from default for confirmable block2 follow-on requests
+GCOAP_RESEND_BUFS_MAX ?= 2
```

Comment:
This value is the count of simultaneous open confirmable requests to support. The default defined in gcoap.h is 1. The CoAP PDU for a confirmable request is stored in `gcoap_state.resend_bufs`. This allows resending the PDU if no response from the server. By default, each buffer uses 128 bytes, so we want to be conservative with how many we allocate.

This value was increased for the gcoap example to support sending the next request for a block2 exchange. The previous request has not yet been deallocated before we allocate the next one. See _resp_handler() in gcoap_cli.c. I could not think of a simple way (asynchronously?) to avoid this extra allocation. I will keep it in mind for future work.

Reply:
Ok, I understand.